test/SlowTests/Issues/RavenDB_15430.cs

```diff
@@ -112,7 +112,7 @@ record = await store.Maintenance.Server.SendAsync(new GetDatabaseRecordOperation
                 record = await store.Maintenance.Server.SendAsync(new GetDatabaseRecordOperation(store.Database));
                 var firstNode2 = record.Topology.Members[0];
 
-                Assert.True(await WaitForValueAsync(async () =>
+                await WaitForValueAsync(async () =>
```

Comment:
If this isn't required to switch, why are we waiting for it?

Reply:
We do want to switch (we want the first in topology to change), but even if it didn't (for example, if the topology has changed regardless in between) - we don't want it to break the test. Does it make sense to you? or should I remove `WaitForValueAsync`?

Reply:
This test explicitly changes the topology, are you saying that this is not needed? Why was it added then by author? If this is required for the test then it makes sense to wait for it to complete, right?

Reply:
@ppekrol It is needed - the `Assert.True` isn't. 
Only the first in topology performs the Rollups and the others get the results through Replication. 
So it makes sense to change the topology just to make things "_harder_"; but I don't see why the test should fail (as in the previous versions of the test). After all, we are not disabling topology updates - then how can we be certain that the topology hasn't changed before?

Reply:
I understand, but action is performed (changing the order) and it basically fails. So maybe the issue is that this fails and the solution is to fix it, not to remove the assert? So wea re either performing a change or not in the test. Having an action without assert that should affect the test is pointless IMO. Why changing the topology fails? To small timeout there? Maybe some cluster issue?

Reply:
This is not a bug, I explored it.. 
- Assume the topology on the first phase of the test (up to line 65) is: {A, B, C}
- Topology changed to: {C, B, A} after line 65.
- The reverse results on lines 117-120 will be: {A, B, C}  -> {C, B, A} -> {A, B, C} -> ... 
   and it fails. 

I can change from Reverse to Random order and expand the timeout.

Reply:
Yes, let's randomize it

Reply:
Done.